pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -770,6 +773,68 @@ func (r *ReconcileHyperConverged) ensureKubeVirtNodeLabellerBundle(instance *hco
 	return r.client.Status().Update(context.TODO(), instance)
 }
 
+func newIMSConfigForCR(cr *hcov1alpha1.HyperConverged, namespace string) *corev1.ConfigMap {
+	labels := map[string]string{
+		"app": cr.Name,
+	}
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "v2v-vmware",
+			Labels:    labels,
+			Namespace: namespace,
+		},
+		Data: map[string]string{
+			"v2v-conversion-image":              os.Getenv("CONVERSION_CONTAINER"),
+			"kubevirt-vmware-image":             os.Getenv("VMWARE_CONTAINER"),
+			"vddk-init-image":                   "kubevirt-vmware/vddk-init:latest",
```

Comment:
This is expected to be changed manually by the cluster admin as the `vddk-init-image` must be created by the end-user (VMware's licence). 

Wouldn't it make more sense to either remove that line (and patch it i.e. by test scripts) or provide more clear "dummy" value as the default? The `kubevirt-vmware/vddk-init:latest` looks valid at the first glance. So without additional knowledge nobody can guess that there is a pitfall here.

Reply:
@mareklibra from talking with @bthurber the idea was that the user would have to push the image to some registry tagged as `kubevirt-vmware/vddk-init:latest` or some other value specified and leaving it up to the registry search to find and pull it.

Talking with @jwmatthews we do have some concerns about using this approach.
- It can lead to downloading the wrong image accidentally (or even maliciously) uploaded to the wrong registry
- Leaving it out and telling the user to add the value to a configmap that is being continually reconciled by an operator seems wrong. Though the operator shouldn't patch out the new value it might lead to the expectation that existing fields can be edited, which would only lead to confusion when they're restored on the next reconcile.

We've asked about the possibility of this just existing in a separate configmap so there is no confusion regarding its existence and what can be edited.

Reply:
> It can lead to downloading the wrong image accidentally (or even maliciously) uploaded to the wrong registry

That's my concern as well. I guess it's more probable that the cluster/repository is "misconfigured" from the kubevirt's perspective than properly querying user-made image.

If nothing better, we need to be sure that at least docker hub, quay and maybe other registries serve **our** dummy content for `kubevirt-vmware/vddk-init:latest`.

I understand, there is intention to simplify the process for the user.
But this approach sounds to me like stopping halfway. From my experience, it can cause more troubles than benefits.